src/bin/main.rs

```diff
@@ -568,6 +605,20 @@ impl GetOptsOptions {
         }
 
         options.config_path = matches.opt_str("config-path").map(PathBuf::from);
+        options.stdin_file_hint = matches.opt_str("stdin-file-hint").map(PathBuf::from);
+
+        // return early if there are issues with the file hint specified
+        if let Some(file_hint) = &options.stdin_file_hint {
+            if !file_hint.exists() {
+                return Err(StdInFileHintError::NotFound(file_hint.to_owned()))?;
+            }
+
+            if let Some(ext) = file_hint.extension() {
+                if ext != "rs" {
```

Comment:
Does rustfmt reject files with other extensions when passed a path? We should probably match that behavior, someone could use a different extension for various reasons.